core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableBatchNestedLoopJoin.java

```diff
@@ -63,9 +64,11 @@ protected EnumerableBatchNestedLoopJoin(
       RexNode condition,
       Set<CorrelationId> variablesSet,
       ImmutableBitSet requiredColumns,
-      JoinRelType joinType) {
+      JoinRelType joinType,
+      Join originalJoin) {
     super(cluster, traits, ImmutableList.of(), left, right, condition, variablesSet, joinType);
     this.requiredColumns = requiredColumns;
+    this.originalJoin = originalJoin;
   }
 
   public static EnumerableBatchNestedLoopJoin create(
```

Comment:
This is public function.

Can we have an old function that be marked as Deprecated?

Reply:
@JiajunBernoulli  we could do that, but in that case if the deprecated function is used, we will have an EnumerableBatchNestedLoopJoin without its originalJoin (so originalJoin would be `null`), what shall we do in that scenario for the RelMdRowCount computation of the EBNLJ?

The alternative would be, we do this as a breaking change to ensure we apply the correction in all cases (we mention on the next release note that "EnumerableBatchNestedLoopJoin now requires on its construction a new parameter: the Join that originated it", tbd before merging the PR). And when downstream projects upgrade, they'd need to adjust their code (if they create EBNLJ on their own).

Both alternatives have pros and cons. Personally I prefer the second one, but I'm open to discuss it.

Reply:
Second is ok.

Let's emphasize this change in JIRA.

Reply:
Ok @JiajunBernoulli .
I have committed the comment on the next RN about the breaking change.
When the PR gets merged, I will mention it also in the Jira.